### ISSUE TITLE
Aggregation query parameter does not replace keys in the lists

### ISSUE THREAD

**Issue by yjcrocks** [pyeve/eve#1025]
I was working on an aggregation endpoint:

```python
things_recommended = {
    'url': 'things/recommended/',
    'datasource': {
        'source': 'things',
        'aggregation': {
            'pipeline': [
                {"$match": {"$or": [{"family": "$family_id"}, {"is_shared": True}]}},
                {"$addFields": {
                    "recommended_for_me":
                    {"$setIsSubset":[["$user_id"], "$recommended_to"]}
                }},
                {"$sort": SON([("recommended_for_me", -1), ("happened", -1)])}
            ]
        }
    }
}
```

However, I found that `$family_id` and `$user_id` are not getting replaced since the code only covers the case of a pure dictionary without lists.

See: https://github.com/pyeve/eve/blob/22ea4bfebc8b633251cd06837893ff699bd07a00/eve/methods/get.py#L126

I'm new to Eve, so I'm not sure that what I found worth an error report. Though it took me quite a time to debug.

Best regards,
Youngjae

PS, Eve is AWESOME.

**Comment by yjcrocks** [comment]
I solved the issue like this:
```python
def parse_aggregation_stage(d, key, value):
        if isinstance(d, dict):
            for st_key, st_value in d.items():
                if isinstance(st_value, dict) or isinstance(st_value, list):
                    parse_aggregation_stage(st_value, key, value)
                if key == st_value:
                    d[st_key] = value
        if isinstance(d, list):
            for st_idx, st_value in enumerate(d):
                if isinstance(st_value, dict) or isinstance(st_value, list):
                    parse_aggregation_stage(st_value, key, value)
                if key == st_value:
                    d[st_idx] = value
```

**Comment by Amedeo91** [comment]
Did you do the PR?

**Comment by zhangtemplar** [comment]
+1 this resolves my problem.

**Comment by zhangtemplar** [comment]
Hi, there is a work around for this problem.

For example, if you have the following end points, which doesn't work as `array` is not yet supported:
```
things_recommended = {
    'url': 'things/recommended/',
    'datasource': {
        'source': 'things',
        'aggregation': {
            'pipeline': [
                {"$match": {"$or": [{"family": "$family_id"}, {"is_shared": True}]}},
            ]
        }
    }
}
```

However, you can overcome this problem by defining the endpoint like:
```
things_recommended = {
    'url': 'things/recommended/',
    'datasource': {
        'source': 'things',
        'aggregation': {
            'pipeline': [
                {"$match": "$where$},
            ]
        }
    }
}
```

The query url is like
```
some_url/things/recommended?aggregate={"$where": {"$or": [{"family": "$family_id"}, {"is_shared": True}]}}
```
Note you need to use `encoderUrlComponent` and `JSON.stringfy` to escape the characters in this url.

I have tested on my computer and it works.

**Comment by aarobc** [comment]
I'm having a similar problem and haven't had much luck. Is the unterminated quote in your example on `{"$match": "$where$},` intentional?

**Comment by zhangtemplar** [comment]
@aarobc sorry, it should be a typo and the correct case would be `{"$match": "$where"}`

**Comment by nicolaiarocci** [comment]
Solved by #1058
